inc/sidebars-emulator.php

```diff
@@ -115,20 +113,50 @@ private function get_widget_option_names( $post_id, $panels_data, $start = 1 ) {
 
 			if( $widget_instance['panels_info']['class'] === 'SiteOrigin_Panels_Widgets_Layout' ) {
 				// Add the widget option names from the layout widget
-				$widget_option_names = array_merge( $widget_option_names, $this->get_widget_option_names( $post_id, $widget_instance[ 'panels_data' ], ++$start ) );
+				$widget_option_names = array_merge( $widget_option_names, $this->get_widget_option_names( $post_id, $widget_instance[ 'panels_data' ] ) );
+			}
+			
+			if ( ! empty( $widget_instance['id'] ) ) {
+				$widget_option_names[] = $widget_instance['option_name'];
 			}
+			$this->all_posts_widgets[ $post_id ][] = $widget_instance;
+		}
 
-			$id_val  = $post_id . strval( ( 10000 * $start ) + intval( $i ) );
+		return $widget_option_names;
+	}
+	
+	/**
+	 * This should be called when a post is saved to set ids required for `is_active_widget` checks. It's necessary to
+	 * do this separately for widgets that call `is_active_widget` in their constructors, e.g. some of Jetpack's widgets
+	 * like Twitter Timeline, Milestone etc.
+	 *
+	 * @param $widgets array The widgets in the layout from $panels_data
+	 * @param $post_id int The post id.
+	 * @param int $start This keeps track of recursive depth.
+	 *
+	 * @return array The widgets array containing updated widgets.
+	 */
+	public function set_widget_ids( $widgets, $post_id, $start = 1 ) {
```

Comment:
This function name is a touch confusing. I don't know if it's just me, but when I see a function name with `set_` I expect it to set something in the class. I'd expect this function to be called `add_widget_ids` or something.

Could just be a personal preference, so this doesn't really need to change.